motr/io_req.c

```diff
@@ -1475,7 +1527,13 @@ static int device_check(struct m0_op_io *ioo)
 
 	M0_LOG(M0_DEBUG, "failed devices = %d\ttolerance=%d", (int)fdev_nr,
 		         (int)layout_k(play));
-	if (is_pver_dud(fdev_nr, layout_k(play), fsvc_nr, max_failures))
+	M0_LOG(M0_DEBUG, "failed services = %d\ttolerance=%d", (int)fsvc_nr,
+			 (int)max_svc_failures);
+	M0_LOG(M0_DEBUG, "failed nodes = %d\ttolerance=%d", (int)fnode_nr,
+			 (int)max_node_failures);
+
+	//TODO: Factor in failed nodes for pver dud calculation.
```

Comment:
This comment will be removed when the TODO is addressed

Reply:
Need to add fnode_nr into pver_dud calculation in a case when node failures are reducing tolerance of the pool version, but is not being accounted for when deciding if pver is dud or not. In a case when 3 nodes have failed, and a drive in 4th node have failed, we have potentially lost 4 units, but the pver is not marked as dud.